source/common/decompressor/zlib_decompressor_impl.cc

```diff
@@ -0,0 +1,68 @@
+#include "common/decompressor/zlib_decompressor_impl.h"
+
+#include "envoy/common/exception.h"
+
+#include "common/common/assert.h"
+
+namespace Envoy {
+namespace Decompressor {
+
+ZlibDecompressorImpl::ZlibDecompressorImpl() : ZlibDecompressorImpl(4096){};
+
+ZlibDecompressorImpl::ZlibDecompressorImpl(uint64_t chunk_size)
+    : chunk_{chunk_size}, initialized_{false}, output_char_ptr_(new unsigned char[chunk_size]),
+      zstream_ptr_(new z_stream(), [](z_stream* z) {
+        inflateEnd(z);
+        delete z;
+      }) {
+  zstream_ptr_->zalloc = Z_NULL;
+  zstream_ptr_->zfree = Z_NULL;
+  zstream_ptr_->opaque = Z_NULL;
+  zstream_ptr_->avail_out = chunk_;
+  zstream_ptr_->next_out = output_char_ptr_.get();
+}
+
+void ZlibDecompressorImpl::init(int8_t window_bits) {
+  ASSERT(initialized_ == false);
+  const int result = inflateInit2(zstream_ptr_.get(), static_cast<int>(window_bits));
```

Comment:
Whats the reason for doing this passing as a int8_t and then casting for args to the library?

Reply:
Not really, I got confused with which type I should use.